src/modules/site-v2/templates/tools/phenotype_database/submit-traits.html

```diff
@@ -130,14 +137,17 @@ <h5 class="modal-title" id="confirmationModalLabel">Thank you</h5>
 
         // If trait is passed as a string value, look up its fields in the phenotype table
         if (trait !== null && typeof trait === 'string') {
-            const el = document.getElementById(`trait-data-${trait}`);
-            trait = {
-                'name':              trait,
-                'dataset':           dataset,
-                'trait_name_caendr': trait,
-                'description_short': el.dataset.description,
-                'species':           el.dataset.species,
-            }
+            queryTraitByName(trait)
+                // Re-call the function using the trait value instead of the name
+                // Response check prevents infinite recursion
+                .then((response) => {
+                    if (response !== null && typeof response === 'string') throw Error(`Invalid response for trait ${id}: "${response}"`);
+                    select_trait(id, response, dataset);
+                })
+                .fail((err) => {
+                    console.error(error);
```

Comment:
should we notify the end-user of the error?

Reply:
Added in [75e9873](https://github.com/AndersenLab/CAENDR/pull/472/commits/75e98735e7c7ebba363679189c0ff72b7eaf1529) -- flashes the error message for the user, like we do elsewhere.